athenahealth/appointment_status.go

```diff
@@ -0,0 +1,57 @@
+package athenahealth
+
+import "strconv"
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+// The athenaNet appointment status. There are several possible statuses.
+// x=cancelled
+// f=future (It can include appointments where were never checked in, even if the appointment date is in the past. It is up to a practice to cancel appointments as a no show when appropriate to do so.)
+// o=open
+// 2=checked in
+// 3=checked out
+// 4=charge entered (i.e. a past appointment)
+type AppointmentStatus string
+
+const (
+	AppointmentStatusCancelled     AppointmentStatus = "x"
+	AppointmentStatusChargeEntered AppointmentStatus = "4"
+	AppointmentStatusCheckedIn     AppointmentStatus = "2"
+	AppointmentStatusCheckedOut    AppointmentStatus = "3"
+	AppointmentStatusFuture        AppointmentStatus = "f"
+	AppointmentStatusOpen          AppointmentStatus = "o"
+)
+
+func (as AppointmentStatus) Int() int {
+	intVal, convErr := strconv.Atoi(string(as))
+	if convErr != nil {
+		return -1
+	}
+
+	return intVal
+}
+
+func (as AppointmentStatus) String() string {
+	if as.Valid() {
+		return string(as)
+	}
+
+	return ""
+}
```

Comment:
I think this method should be as simple as `return string(as)` since the caller is still required to validate the appointment status by checking if the string is empty. We're not saving anything in the way of interacting with this by doing these extra steps. Infact, we're adding extra steps in type converting for values that we know are valid i.e. storing after already having validated.